### PR TITLE
Bug 1129584 - initialize MediaDB and start creating thumbnails sooner

### PR DESCRIPTION
**Pull request by davidflanagan**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[davidflanagan David Flanagan (davidflanagan)](https://github.com/davidflanagan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5b06baf7bbc8)

**Comment by try-server-hook** [comment]
[davidflanagan David Flanagan (davidflanagan)](https://github.com/davidflanagan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=d2ccf9745c16)

File: apps/gallery/js/thumbnails.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      completionResolver = resolve;
+
+      // Now that the promises are ready, start creating thumbnails
+      createThumbnails();
+    });
+  });
+
+  function createThumbnails() {
+    // If the db is ready, start enumerating it. Otherwise wait
+    // until it becomes ready
```
Comment (punamdahiya) [inline apps/gallery/js/thumbnails.js]: nit: Should comments here say if the db is ready or can be enumerated (i.e. MediaDB.ENUMERABLE), start enumerating it

**Comment by try-server-hook** [comment]
[davidflanagan David Flanagan (davidflanagan)](https://github.com/davidflanagan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=19f46668e932)

**Pull request merged**